lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -482,14 +502,154 @@ namespace Intl
         error = UErrorCode::U_ZERO_ERROR;
 
         written = uloc_toLanguageTag(defaultLocaleId, bcp47, ULOC_FULLNAME_CAPACITY, true, &error);
-        ICU_ASSERT_RETURN(error, written > 0 && written < cchLangtag, 0);
+        ICU_ASSERT_RETURN(error, written  > 0 && ((size_t) written) < cchLangtag, 0);
 
         return utf8::DecodeUnitsIntoAndNullTerminateNoAdvance(
             langtag,
             reinterpret_cast<LPCUTF8>(bcp47),
             reinterpret_cast<LPCUTF8>(bcp47 + written)
         );
     }
+
+    bool CollatorGetResolvedLocale(_In_z_ const char *langtag, _In_ size_t cchLangtag, _Out_ char *baseLangtag, _In_ size_t cchBaseLangtag,
+        _Out_ char *collation, _In_ size_t cchCollation)
+    {
+        // ensure the buffers have allocated memory and that their lengths can be safely casted to int32_t
+        Assert(cchBaseLangtag > 0 && cchBaseLangtag < INT_MAX && cchCollation > 0 && cchCollation < INT_MAX);
```

Comment:
While we're asserting things, should add assert that out parameters are not `nullptr`.

Reply:
Good call